spec/paperclip/storage/s3_spec.rb

```diff
@@ -1272,15 +1438,23 @@ class DummyCredentialProvider; end
 
     it "runs the file through ERB" do
       assert_equal 'env_bucket', @dummy.avatar.bucket_name
-      assert_equal 'env_key', @dummy.avatar.s3_bucket.config.access_key_id
-      assert_equal 'env_secret', @dummy.avatar.s3_bucket.config.secret_access_key
+
+      assert_equal 'env_key',
+        (defined?(::Aws) ?
+          @dummy.avatar.s3_bucket.client.config.access_key_id :
+          @dummy.avatar.s3_bucket.config.access_key_id)
+
+      assert_equal 'env_secret',
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.